lib/galaxy/datatypes/binary.py

```diff
@@ -126,6 +126,34 @@ def sniff(self, filename):
             return False
 
 
+class Gpr (Binary):
+    """ Gpr File format described at:
+    http://mdc.custhelp.com/app/answers/detail/a_id/18883/kw/18883/session/L2F2LzEvdGltZS8xNTUxMjY5NTc2L3NpZC9MajhQX3E4bw%3D%3D#gpr
+    """
+    edam_format = "format_3829"
+    edam_data = "data_3110"
+    file_ext = "gpr"
+
+    def sniff(self, filename):
```

Comment:
I'd suggest you to use `sniff_prefix(self, file_prefix)` instead. Then you can deal with the already opened file `file_prefix` (of type `FilePrefix`), which should deal well with UTF-8 characters and has various convenient methods. See `lib/galaxy/datatypes/molecules.py` for many examples.
Please note that you would need to decorate your class with `@build_sniff_from_prefix` in that case.

Reply:
The problem is UTF-8 as shown in my previous message.

Reply:
To clarify my message above, I think you are seeing the UTF-8 error because of how you are opening the file. The appropriate way would be either:
```python
            source = io.open(filename, 'r', encoding='utf-8').read(2000)
```
or changing this method to `sniff_prefix()`, as described above.